### PR TITLE
Separate out Unix string parsing logic and test it

### PR DESCRIPTION
**Pull request by mellinoe**
These changes separate out all of the string parsing logic into a separate class, which I have added unit tests for. The unit tests operate on some static test files that I have created / copied from my dev machine. They are mostly unchanged, but I have added some more interesting data to the files in some cases.

This should protect us from breaks in the parsing logic in the future, as we are using quite a bit of shared, common code that might be changed often in the future. I discovered two issues during the writing of these tests. One was that the Linux parsing logic for current active/listening TCP and UDP sockets was using the wrong entry in the table. It was using the remote endpoint rather than the local endpoint as listed in the file. The other was that the samba config file had an incorrect path (smb/smb.conf rather than samba/smb.conf)

NOTE: Most of the diff here is just moving the string parsing logic out of the Linux network classes, and into a few partial files for the StringParsingHelpers class. Looking at the unit tests first will give a good idea of how the parsing helpers are used. There shouldn't be any functional difference here in the source code other than for the two issues I discovered above.

Addresses #4059.

@CIPop , @vijaykota ,

### CODE REVIEW

**Comment by bbowyersmyth** [comment]
I see that this is mainly moving of existing code but there are quite a few cases where the default overload of `string.indexOf(string, ...) is being used. It doesn't look like most of them are intended to be CurrentCulture based comparisons.

**Comment by mellinoe** [comment]
@bbowyersmyth That's a good point. They should probably all be switched to invariant checks.

**Comment by mellinoe** [comment]
@dotnet-bot Test this, please

(Having some issues with different line endings in the test files, although I can't repro on my linux machine...)

File: src/System.Net.NetworkInformation/tests/FunctionalTests/NetworkFiles/dev [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Inter-|   Receive                                                |  Transmit
```
Comment (CIPop) [inline src/System.Net.NetworkInformation/tests/FunctionalTests/NetworkFiles/dev]: My understanding was that we shouldn't add any test data in this repo. Instead you'll need to use the test-data repo and create a Nuget Package containing these files.

**Comment by CIPop** [comment]
LGTM with the exception of the test data files that shouldn't be in this partition. @bartonjs previously created a nuget package for the test certificates. For an example you can see how we're using it in System.Net.Security Functional tests.

**Comment by mellinoe** [comment]
The test assets are just tiny plain text files, so I don't think it makes sense to separate them out. AFAIK we are only storing test assets elsewhere when they are prohibitively large, or for things like binaries, zip files, etc. I guess that test certificates fall under that category (large binary blobs). The corefx-test repo has such things in it, but I don't think we should put these there.

I was hitting some issues in CI because the test files were copied over from a Windows machine, which meant they had different line endings than if they were git-cloned from a Unix machine. I thought this would happen and had some normalization code to fix it, but my logic was messed up. It should be fixed now, so CI should pass.

**Comment by ellismg** [comment]
Why did the project.lock.json file change if there is no corresponding project.json change?

**Comment by mellinoe** [comment]
Not sure about that lock file. I manually reset it and it didn't change again.

**Comment by mellinoe** [comment]
@dotnet-bot test Code Formatter Check

**Comment by mellinoe** [comment]
I haven't changed any native code here, so I'm going to merge this in anyways, as the Code Formatter Check is failing for all of our PR's.

I've added a bullet to my issue list about the string culture comparisons and will address that separately.

**Pull request merged**